code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -529,10 +530,11 @@
 
 	if(W.tool_behaviour == TOOL_WRENCH)
 		if(length(all_components))
-			to_chat(user, span_warning("Remove all components from \the [src] before disassembling it."))
+			user.balloon_alert(user, "\the [src] still has components to remove")
 			return
+		W.play_tool_sound(src, user, 20, volume=20)
 		new /obj/item/stack/sheet/iron( get_turf(src.loc), steel_sheet_cost )
-		physical.visible_message(span_notice("\The [src] is disassembled by [user]."))
+		user.balloon_alert(user,"you disassemble \the [src]")
```

Comment:
```suggestion
		user.balloon_alert(user, "disassembled")
```

Reply:
Apologies if I'm being annoying, I really do want to learn. Why would we use user.balloon_alert on this one specifically?

Reply:
Because this is where it gets deconstructed, so `src` is going to be different.

Reply:
Theoretically you would call `balloon_alert` on the turf, but there's a bug in that I haven't fixed yet.